src/python/pants/backend/python/rules/coverage/plugin.py

```diff
@@ -0,0 +1,97 @@
+# coding=utf-8
```

Comment:
Is this identical to the V1 plugin? If so, is it possible to have the V1 `pytest_run.py` code point to this plugin and delete the V1 one so that we don't have the duplication?

Reply:
Oh, actually, this V2 coverage implementation _must_ use Coverage 5+, right? Which implies it only works with Python 3? If this is the case, this plugin can be rewritten to use Python 3 :D 

This means:
- remove `__future__` imports
- use `super()` rather than the more verbose form
- +1 for type hints (can punt on this)
- +1 for `pathlib.Path` (can punt on this)

IMO, we'd want to have this Python-3 only implementation for V2 to represent the future of the plugin. Then, we'd keep the original Python 2 one where it as for legacy support.